ugt_fwtools/synth_1_module.py

```diff
@@ -0,0 +1,74 @@
+import re
+import argparse
+import configparser
+import logging
+import os
+import subprocess
+from . import utils
+from .synthesis import create_module, implement_module, show_screen_sessions, start_screen_session
+
+def parse_args():
+    """Parse command line arguments."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument("mod_id", help="module number (eg. 1)")
+    parser.add_argument("path", metavar="<path>", type=os.path.abspath, help=f"fw build path to config file")
+    return parser.parse_args()
+
+def main() -> None:
+    """Main routine."""
+
+    # Parse command line arguments.
+    args = parse_args()
+    
+    if not os.path.isfile(args.path):
+        raise RuntimeError(
+            f"\033[1;31m no such file {args.path!r} \033[0m"
+        )
+
+    config = configparser.ConfigParser()
+    config.read(args.path)
+
+    args.board_type = config.get("device","name")
+    args.build = config.get("menu","build")
+    args.ipbb_dir = config.get("firmware","buildarea")
+    args.project_type = config.get("firmware","type")
+    args.vivado = config.get("vivado","version")
+    module_id = args.mod_id
+    module_name = f"module_{module_id}"
+    module_path = os.path.join(args.ipbb_dir, "proj", module_name)
+
+    # Check for UGT_VIVADO_BASE_DIR
+    args.vivado_base_dir = os.getenv("UGT_VIVADO_BASE_DIR")
+    if not args.vivado_base_dir:
+        raise RuntimeError("\033[1;31m Environment variable 'UGT_VIVADO_BASE_DIR' not set. Set with: 'export UGT_VIVADO_BASE_DIR=...' \033[0m")
+
+    # Vivado settings
+    args.settings64 = os.path.join(args.vivado_base_dir, args.vivado, "settings64.sh")
+    if not os.path.isfile(args.settings64):
+        raise RuntimeError(
+            f"\033[1;31m no such Xilinx Vivado settings file {args.settings64!r}\n \033[0m"
+            f"   \033[1;31m check if Xilinx Vivado {args.vivado} is installed on this machine. \033[0m"
```

Comment:
Please try avoid using[ ANSI escape codes](https://en.wikipedia.org/wiki/ANSI_escape_code) in exception messages, better to separate concerns; exception messages should contain concise information about an error while ANSI codes can be used by some TTYs to format the output. As the output is not automatically a TTY (like a pipe to a text file, log file, other function handling an exception) applying ANSI escape codes should be applied only when testing `sys.stdout.isatty()` or by using a library for handling colored output like [coloredlogs](https://pypi.org/project/coloredlogs/).

Reply:
A simple solution without dependencies could be...
```python
tty_bold_red = "\033[1;31m"
tty_reset = "\033[0m"

def print_error(message):
    if sys.stdout.isatty():
        message = f"{tty_bold_red}{message}{tty_clear}"
    print(message)
...
if error_condition:
    message = "my error description"
    print_error(message):  # will print in red on a TTY
    raise RuntimeError(message)  # can be handled as supposed
```

Reply:
done!